### PR TITLE
[WIP] Update all dependencies (including nginx)

### PR DESCRIPTION
**Pull request by datWeazel**
I marked the PR as WIP since I couldn't test it yet.

This PR updates the used nginx base image to 1.26.1.
Since the last update some things changed that's why I made some changes.

- /bin is now a symlink to /usr/bin
Since `/bin` is now a symlink to `/usr/bin` I moved the `fs_overlay/bin` to `fs_overlay/usr/bin` so the files are copied correctly.

- Changed the repository of `docker-gen`
`docker-gen` is no longer in the personal github of `jwilder` but in `nginx-proxy` (https://github.com/nginx-proxy/docker-gen). I updated the path and updated the dependency since the release that was currently used was from 2018.

- Updated `s6-overlay`
The used `s6-overlay` version was from 2020, and since a already introduced some major changes I updated it aswell.

### CODE REVIEW

**Comment by datWeazel** [comment]
image

https-portal seems to be working now with latest dependencies. Changes were more substantial than I initially thought but I hope I got everything now.

**Comment by SteveLTN** [comment]
Thanks you very much!

I'll give it a test (probably over the weekend).

File: fs_overlay/etc/cont-init.d/00-welcome [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 echo '
 ========================================
-HTTPS-PORTAL v1.23.1
+HTTPS-PORTAL v1.26.1
```
Comment (SteveLTN) [inline fs_overlay/etc/cont-init.d/00-welcome:5]: HTTPS-PORTAL version does not really follow NGINX version. I will later change it to 1.24.

**Review by SteveLTN — APPROVED** [review]

**Comment by SteveLTN** [comment]
Thanks for the great work! I'll merge it now and give it a test, maybe make some small changes (such as the version number as I mentioned in the comment), then release a new version.

**Comment by SteveLTN** [comment]
@datWeazel I merged an encouterred some errors while trying to build with `make docker-multiarch`. This is to build the image in multiple architecture.

```
87.02 Get:66 http://deb.debian.org/debian bookworm/main arm64 unzip arm64 6.0-28 [157 kB]
87.04 Get:67 http://deb.debian.org/debian bookworm/main arm64 zip arm64 3.0-13 [223 kB]
91.35 debconf: delaying package configuration, since apt-utils is not installed
92.05 Fetched 23.4 MB in 3s (7521 kB/s)
92.34 Error while loading /usr/sbin/dpkg-split: No such file or directory
92.35 Error while loading /usr/sbin/dpkg-deb: No such file or directory
92.36 dpkg: error processing archive /var/cache/apt/archives/cron-daemon-common_3.0pl1-162_all.deb (--unpack):
92.36  dpkg-deb --control subprocess returned error exit status 1
92.38 Error while loading /usr/sbin/dpkg-split: No such file or directory
92.39 Error while loading /usr/sbin/dpkg-deb: No such file or directory
92.39 dpkg: error processing archive /var/cache/apt/archives/sensible-utils_0.0.17+nmu1_all.deb (--unpack):
92.39  dpkg-deb --control subprocess returned error exit status 1
92.45 Errors were encountered while processing:
92.45  /var/cache/apt/archives/cron-daemon-common_3.0pl1-162_all.deb
92.46  /var/cache/apt/archives/sensible-utils_0.0.17+nmu1_all.deb
92.83 E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Google around and the solution does not seem to work. I'm still investigating.

**Comment by datWeazel** [comment]
@SteveLTN One issue I found: s6-overlay package for x86-64 seems to be  `s6-overlay-x86_64.tar.xz` now. `archname` currently resolves it as `amd64` so the Archive can't be loaded. `x86` seems to be dropped completely, so can't build for `linux/386` either.

I updated the Makefile to drop building for `linux/386` and use the correct s6-overlay package.
Since this is already merged the changes are currently only in my fork. No idea how to push the changes without opening a new PR. 

https://github.com/datWeazel/https-portal/commit/e5961cef1e97ab89515fc20aef4299d9e476da6f

**Comment by SteveLTN** [comment]
Yes. I found that in the latest version of s6-overlay release page, the arch names have been changed. And I modified [archname](https://github.com/SteveLTN/https-portal/blob/master/fs_overlay/usr/bin/archname) script to fix this issue. 

But the harder problem to solve was something related to `docker buildx`. When building for the `arm64` arch, I kept getting these two error messages:

```
92.34 Error while loading /usr/sbin/dpkg-split: No such file or directory
92.35 Error while loading /usr/sbin/dpkg-deb: No such file or directory
```

Googling it found that it might be an issue with the `multiarch/qemu-user-static` image. But building on a remote x86 server seemed to solve this issue (I'm on an Apple Silicon Mac). 

Just need to do more test now.

**Comment by SteveLTN** [comment]
I think everything is working now. I'm putting it on test at https://https-portal-test.steveltn.me/, Will let it run for 1-2 days (to test the auto-renewal, as I set the renew frequency to be once a day for myself). After that, I'll make a new release.

**Pull request merged**